plugins/bigcommerce/src/plugin.ts

```diff
@@ -0,0 +1,153 @@
+import { registerCommercePlugin } from '@builder.io/commerce-plugin-tools';
+import pkg from '../package.json';
+
+registerCommercePlugin(
+  {
+    name: 'BigCommerce',
+    id: pkg.name,
+    settings: [
+      {
+        name: 'storeHash',
+        type: 'string',
+        required: true,
+      },
+      {
+        name: 'accessToken',
+        type: 'string',
+        required: true,
+      },
+    ],
+    ctaText: `Connect your BigCommerce store`,
+  },
+  settings => {
+    const basicCache = new Map();
+
+    const storeHash = settings.get('storeHash');
+    const accessKey = settings.get('accessToken');
+
+    const baseUrl = (url: string) => {
+      const endUrl = `https://api.bigcommerce.com/stores/${storeHash}/v3/catalog/${url}`;
+      return `https://builder.io/api/v1/proxy-api?url=${encodeURIComponent(endUrl)}`;
+    };
+
+    const headers = {
+      'X-Auth-Token': accessKey,
+      Accept: 'application/json; charset=utf-8',
+      'Content-Type': 'application/json',
+    };
+
+    const transformResource = (resource: any) => ({
+      ...resource,
+      id: resource.id,
+      title: resource.name,
+      handle:
+        resource.customUrl?.url,
+      image: {
+        src: resource.primary_image?.url_thumbnail || resource.image_url
+      },
+    });
+
+    const service = {
+      product: {
+        async findById(id: string) {
+          const key = `${id}productById`;
+          // https://api.bigcommerce.com/stores/${storeHash}/v3/catalog/products/${product_id}
+          const product =
+            basicCache.get(key) ||
+            (await fetch(baseUrl(`products/${id}?include=primary_image`), { headers })
+              .then(res => res.json())
+              .then(transformResource));
+          basicCache.set(key, product);
+          return product;
+        },
+
+        /* async findByHandle(handle: string) {
```

Comment:
although this is optional in the interface it's still needed for all product/collection resources

Reply:
expanding on the need for handles (slugs / url_keys), it allows the developer to read the handle from url params and send it as custom targeting attributes in one api call , as opposed to IDs, where they'll need to fetch the product object from the `handle` in url and they send the ID to builder for targeting

Reply:
So Bigcommerce doesn't have an API to fetch by handle. Instead, I've just done a get all products, and searched for a matching product with the handle. Not sure if I should deal with pagination use-cases, where it might not  be in the first 100 results. Whats the recommendation for those usecases?